src/h3lib/lib/h3Index.c

```diff
@@ -446,12 +446,12 @@ int H3_EXPORT(h3IsPentagon)(H3Index h) {
  * @param h The H3Index.
  * @return The highest resolution non-zero digit in the H3Index.
  */
-int _h3LeadingNonZeroDigit(H3Index h) {
+Direction _h3LeadingNonZeroDigit(H3Index h) {
     for (int r = 1; r <= H3_GET_RESOLUTION(h); r++)
         if (H3_GET_INDEX_DIGIT(h, r)) return H3_GET_INDEX_DIGIT(h, r);
```

Comment:
Nit: As above, this requires knowledge that the center digit is 0. Maybe slightly more semantic to rewrite this as `_h3LeadingNonCenterDigit` and compare `if (H3_GET_INDEX_DIGIT(h, r) !== CENTER_DIGIT)`? No strong feeling though, it might not be possible/desirable to fully abstract away the concrete implementation.